rasterio/vrt.py

```diff
@@ -245,3 +245,57 @@ def _boundless_vrt_doc(
         dstrect.attrib['ySize'] = str(src_dataset.height)
 
     return ET.tostring(vrtdataset, encoding="utf-8")
+
+
+class VirtualDataset:
+    """A lazily evaluated virtual dataset based on GDAL's VRT."""
+
+    @classmethod
+    def fromstring(cls, text):
+        """Create a virtual dataset from XML.
+
+        Paramters
+        ---------
+        text : bytes or str
+            XML as a byte or unicode string.
+
+        Returns
+        -------
+        VirtualDataset
+
+        """
+        # To avoid a circular import.
+        from rasterio.io import MemoryFile
```

Comment:
:see_no_evil:

Reply:
Would it make sense to put VirtualDataset with the rest of the Dataset objects in rasterio.io?